src/workers/_inputs.js

```diff
@@ -78,6 +91,48 @@ function loadMatrixMarketRaw(files) {
     }
 
     permuteGenes(cache.genes);
+
+    var annotations_file = files.filter(x => x.type == "annotations");
+    if (annotations_file.length == 1) {
+        var annotations_file = annotations_file[0]
+        var content = new Uint8Array(annotations_file.buffer);
+        var ext = annotations_file.name.split('.').pop();
+        let compressed = false;
+
+        if (ext == "gz") {
```

Comment:
Could also shift this logic into `read(C)SVFromBuffer` if a file name is supplied instead of `compressed`.